### PR TITLE
Make searchbar an extension point

### PR DESCRIPTION
**Pull request by MarcinAman**

### CODE REVIEW

File: plugins/base/src/main/kotlin/renderers/html/htmlPreprocessors.kt [hunk 1]
```diff
@@ -13,22 +13,24 @@ import org.jetbrains.dokka.plugability.DokkaContext
 import org.jetbrains.dokka.plugability.configuration
 import org.jetbrains.dokka.transformers.pages.PageTransformer
 
-object NavigationPageInstaller : PageTransformer {
+open class NavigationPageInstaller : PageTransformer {
```
Comment (MarcinAman) [inline plugins/base/src/main/kotlin/renderers/html/htmlPreprocessors.kt:16]: I have some doubts it this should have a separate interface

Comment (sellophane) [reply plugins/base/src/main/kotlin/renderers/html/htmlPreprocessors.kt:16]: Yes.

File: plugins/base/src/main/kotlin/DokkaBase.kt [hunk 2]
```diff
@@ -207,4 +208,8 @@ class DokkaBase : DokkaPlugin() {
             MultimodulePageCreator(it)
         }
     }
+
+    val baseSearchbarDataInstaller by extending {
```
Comment (sellophane) [inline plugins/base/src/main/kotlin/DokkaBase.kt:212]: It's acceptable solution, but my personal preference would be to have this as yet another htmlPreprocessor, which adds to pages tree a new, special page, let's call it `PlainJsPage`, with contents from `searchbarDataInstaller.generatePagesList()` and renders it in HtmlRenderer as `outputWriter.write("scripts/pages", (page as PlainJsPage).content, ".js")`

This omits the need of having another extension point for this particular reason and is more consistent with how we generate similar pages (Index, Navigation, etc)

File: plugins/base/src/main/kotlin/renderers/html/SearchbarDataInstaller.kt [hunk 3]
```diff
@@ -4,74 +4,107 @@ import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
...
+    data class PageWithId(val id: PageId, val page: ContentPage) {
+        val displayableSignature = getSymbolSignature(page)?.let { flattenToText(it) } ?: page.name
+    }
 
-    private val pageList = ConcurrentHashMap<String, Pair<String, String>>()
+    private val mapper = jacksonObjectMapper()
 
-    open fun generatePagesList(): String {
-        val pages = pageList.entries
+    open fun generatePagesList(pages: Map<PageId, PageWithId>, locationResolver: PageResolver): Json =
```
Comment (pikinier20) [inline plugins/base/src/main/kotlin/renderers/html/SearchbarDataInstaller.kt:32]: I think returning there Json implies that you need to use jackson. String was more flexible

**Review by pikinier20 — COMMENTED** [review]
I think it's OK, just take a look at my previous comment there

File: plugins/base/src/main/kotlin/renderers/html/SearchbarDataInstaller.kt (hunk 3 above)
Comment (MarcinAman) [reply plugins/base/src/main/kotlin/renderers/html/SearchbarDataInstaller.kt:32]: I don't think it does since JSON is just an alias for String

**Review by kamildoleglo — APPROVED** [review]

**Pull request merged**